### PR TITLE
Implement centered and justified justifications

### PR DESCRIPTION
**Pull request by ExcaliburZero**
Implement centered and justified text justifications.

Also add some documentation to related functions and types.

### CODE REVIEW

File: Console/Display.hs [hunk 1]
```diff
@@ -256,20 +260,83 @@ summarySet :: Summary -> [OutputElem] -> IO ()
...
+    | otherwise          = justifiedString
   where
-    szS = length s
+    justifiedString  = intercalate spaces $ insertExcessSpaces excessChars (words string)
+    spaces           = replicate spacing ' '
+    excessChars      = lengthDifference `mod` (numberOfWords - 1)
+    spacing          = lengthDifference `div` (numberOfWords - 1)
+    lengthDifference = size - wordsLength
+    wordsLength      = sum $ map length $ words string
+    numberOfWords    = length $ words string
```
Comment (NicolasDP) [inline Console/Display.hs]: `words string` is use 3 times on the same `string`.

**Comment by NicolasDP** [comment]
Nice, looking good. Just a minor change needed.

File: Console/Display.hs (hunk 1 above)
Comment (ExcaliburZero) [inline Console/Display.hs]: Okay, fixed.

**Comment by NicolasDP** [comment]
thanks for the fixes

**Pull request merged**